dts/common/nordic/nrf54l20.dtsi

```diff
@@ -0,0 +1,614 @@
+			clock-frequency = <DT_FREQ_M(128)>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			itm: itm@e0000000 {
+				compatible = "arm,armv8m-itm";
+				reg = <0xe0000000 0x1000>;
+				swo-ref-frequency = <DT_FREQ_M(128)>;
+			};
+		};
+
+		cpuflpr: cpu@1 {
+			compatible = "nordic,vpr";
+			reg = <1>;
+			device_type = "cpu";
+			clock-frequency = <DT_FREQ_M(128)>;
+			riscv,isa = "rv32emc";
+			nordic,bus-width = <32>;
+		};
+	};
+
+	clocks {
+		lfxo: lfxo {
+			compatible = "nordic,nrf-lfxo";
+			#clock-cells = <0>;
+			clock-frequency = <32768>;
+		};
+
+		hfxo: hfxo {
+			compatible = "nordic,nrf-hfxo";
+			#clock-cells = <0>;
+			clock-frequency = <DT_FREQ_M(32)>;
+		};
+	};
+
+	soc {
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+#ifdef USE_NON_SECURE_ADDRESS_MAP
+		/* intentionally empty because UICR is hardware fixed to Secure */
+#else
+		uicr: uicr@ffd000 {
+			compatible = "nordic,nrf-uicr";
+			reg = <0xffd000 0x1000>;
+		};
+#endif
+		ficr: ficr@ffc000 {
+			compatible = "nordic,nrf-ficr";
+			reg = <0xffc000 0x1000>;
+			#nordic,ficr-cells = <1>;
+		};
+
+		cpuapp_sram: memory@20000000 {
+			compatible = "mmio-sram";
+			reg = <0x20000000 DT_SIZE_K(188)>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x20000000 0x2f000>;
+		};
+
+		cpuflpr_sram: memory@2002f000 {
+			compatible = "mmio-sram";
+			reg = <0x2002f000 DT_SIZE_K(68)>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x2002f000 0x11000>;
+		};
+
+#ifdef USE_NON_SECURE_ADDRESS_MAP
+		global_peripherals: peripheral@40000000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x40000000 0x10000000>;
+#else
+		global_peripherals: peripheral@50000000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x50000000 0x10000000>;
+#endif
+
+			dppic00: dppic@42000 {
+				compatible = "nordic,nrf-dppic";
+				reg = <0x42000 0x808>;
+				status = "disabled";
+			};
+
+			spi00: spi@4d000 {
+				/*
+				 * This spi node can be either SPIM or SPIS,
+				 * for the user to pick:
+				 * compatible = "nordic,nrf-spim" or
+				 *              "nordic,nrf-spis".
+				 */
+				compatible = "nordic,nrf-spim";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <0x4d000 0x1000>;
+				interrupts = <77 NRF_DEFAULT_IRQ_PRIORITY>;
+				max-frequency = <DT_FREQ_M(32)>;
+				easydma-maxcnt-bits = <16>;
+				rx-delay-supported;
+				rx-delay = <1>;
+				status = "disabled";
+			};
+
+			uart00: uart@4d000 {
+				compatible = "nordic,nrf-uarte";
+				reg = <0x4d000 0x1000>;
+				interrupts = <77 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+
+			cpuflpr_vpr: vpr@4c000 {
+				compatible = "nordic,nrf-vpr-coprocessor";
+				reg = <0x4c000 0x1000>;
+				ranges = <0x0 0x4c000 0x1000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				status = "disabled";
+
+				cpuflpr_clic: interrupt-controller@f0000000 {
+					compatible = "nordic,nrf-clic";
+					reg = <0xf0000000 0x1780>;
```

Comment:
Is this correct? If I'm not mistaken, this property is an offset from the `ranges` of its parent (0x4004c000) what gives 0x13004c000 - out of addressable memory.

I think clic should be defined outside of `global_peripherals` in a FLPR-specific file, as an IRQ controller at the `/soc/` level.

Reply:
As it is not supported for now, I will remove it